### DISCUSSION TITLE
R shiny: Dynamic UI & Server Parameter/Argument

### DISCUSSION THREAD

**Discussion by naturallyfunny** [teamalgoritma/community#769] [Q&A] [answered]
## Pertanyaan:
1. Bagaimana cara memberi argumen/parameter untuk server (output$plot_cluster)? saya ingin memberi argumen variable _**words**_ dari UI agar bisa digunakan oleh server (output$cluster).

2. Bagaimana cara menghasilkan box dengan jumlah yang dinamis? Saya ingin menghasilkan box dalam jumlah yang berdasarkan input nilai _**k**_. Saya sudah coba menggunakan for loops namun sepertinya tidak berhasil.

## Plot Function from Server:
 ```
output$plot_cluster <- renderPlot({
    req(input$cluster)
    
    # Ini function pribadi saya
    get_WordCloud(words)
  })
```

  ## UI from Server:
```
  output$second_row <- renderUI({
    req(input$cluster)
    
    if (is.null(input$resumes)) {
      # Bagian ini tidak perlu diperhatikan. Tidak ada hubungan dengan pertanyaan
      )
    }
    else{
      k <- input$n_cluster
      ranking <- get_ranking_for_cluster()
      cluster <- get_cluster(ranking, k)
      cluster_strings <- get_clusters_strings(cluster, ranking)
      
      # ----2nd row----
      width <- 12/k
      count <- 0
      fluidRow(
        for (words in cluster_strings){
          count <- count + 1
          box(
            width = width,
            label = paste("Cluster ", count, ":", sep = ""),
            plotOutput("plot_cluster")
          )
        }
      )
    }
  })
```

**Comment by divakartika** [comment]
Dear @ArdiantheGreat 

Pertanyaan Anda telah kami terima dan sedang kami usahakan terkait solusinya. Mohon kesediaannya untuk menunggu, terima kasih.

Best regards,
Team Algoritma

**Reply by naturallyfunny** [reply, marked as answer]
Terima kasih sebelumnya, namun saya sudah menemukan solusinya.
